packages/plugin-dts/src/utils.ts

```diff
@@ -95,10 +95,17 @@ export async function addBannerAndFooter(
   const content = await fsP.readFile(file, 'utf-8');
   const code = new MagicString(content);
 
-  banner && code.prepend(`${banner}\n`);
-  footer && code.append(`\n${footer}\n`);
+  if (banner && !content.trimStart().startsWith(banner.trim())) {
+    code.prepend(`${banner}\n`);
+  }
+
+  if (footer && !content.trimEnd().endsWith(footer.trim())) {
```

Comment:
When will the code end with the footer while still trying to append it again?

Reply:
In watch mode of incremental build, only save code with no change will cause an incremental dts emit, which will trigger this fuction of writing banner and footer. It is different from watch mode of no incremental build...

Reply:
In watch mode of incremental build, only save code with no change will cause an incremental dts emit, which will trigger this fuction of writing banner and footer. It is different from watch mode of no incremental build...